pkg/proxy/kafka.go

```diff
@@ -159,10 +178,192 @@ func (k *kafkaRedirect) canAccess(req *kafka.RequestMessage, numIdentity policy.
+	info.LabelsSHA256 = r.conf.source.GetLabelsSHA()
+	info.Identity = uint64(r.conf.source.GetIdentity())
+	r.conf.source.RUnlock()
+}
+
+// getInfoFromConsumable fills the accesslog.EndpointInfo fields, by fetching
+// the consumable from the consumable cache of endpoint using identity sent by
+// source. This is needed in ingress proxy while logging the source endpoint
+// info.  Since there will be 2 proxies on the same host, if both egress and
+// ingress policies are set, the ingress policy cannot determine the source
+// endpoint info based on ip address, as the ip address would be that of the
+// egress proxy i.e host.
+func (r *kafkaRedirect) getInfoFromConsumable(ipstr string,
+	info *accesslog.EndpointInfo,
+	srcIdentity policy.NumericIdentity) {
+	ep := r.conf.source
+	ip := net.ParseIP(ipstr)
+	if ip != nil {
+		if ip.To4() != nil {
+			info.IPv4 = ip.String()
+		} else {
+			info.IPv6 = ip.String()
+		}
+	}
+	secIdentity := ep.ResolveIdentity(srcIdentity)
+
+	if secIdentity != nil {
+		info.Labels = secIdentity.Labels.GetModel()
+		info.LabelsSHA256 = secIdentity.Labels.SHA256Sum()
+		info.Identity = uint64(srcIdentity)
+	}
+}
+
+func (r *kafkaRedirect) getSourceInfo(remoteAddr string,
+	srcIdentity policy.NumericIdentity) (accesslog.EndpointInfo,
+	accesslog.IPVersion) {
+	info := accesslog.EndpointInfo{}
+	version := accesslog.VersionIPv4
+	ipstr, port, err := net.SplitHostPort(remoteAddr)
+	if err == nil {
+		p, err := strconv.ParseUint(port, 10, 16)
+		if err == nil {
+			info.Port = uint16(p)
+		}
+
+		ip := net.ParseIP(ipstr)
+		if ip != nil && ip.To4() == nil {
+			version = accesslog.VersionIPV6
+		}
+	}
+
+	// At egress, the local origin endpoint is the source
+	if !r.ingress {
+		r.localEndpointInfo(&info)
+	} else if err == nil {
+		if srcIdentity != 0 {
+			r.getInfoFromConsumable(ipstr, &info, srcIdentity)
+		} else {
+			// source security identity 0 is possible when somebody else
+			// other than the BPF datapath attempts to
+			// connect to the proxy.
+			// We should log no source information in that case, in the proxy log.
+			log.Warn("Missing security identity in source endpoint info")
+		}
+
+	}
+
+	return info, version
+}
+
+// fillReservedIdentity resolves the labels of the specified identity if known
+// locally and fills in the following info member fields:
+//  - info.Identity
+//  - info.Labels
+//  - info.LabelsSHA256
+func (r *kafkaRedirect) fillReservedIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
+	info.Identity = uint64(id)
+
+	if c := policy.GetConsumableCache().Lookup(id); c != nil {
+		if c.Labels != nil {
+			info.Labels = c.Labels.Labels.GetModel()
+			info.LabelsSHA256 = c.Labels.GetLabelsSHA256()
+		}
+	}
+}
+
+// egressDestinationInfo returns the destination EndpointInfo for a flow
+// leaving the proxy at egress.
+func (r *kafkaRedirect) egressDestinationInfo(ipstr string, info *accesslog.EndpointInfo) {
+	ip := net.ParseIP(ipstr)
+	if ip != nil {
```

Comment:
To avoid indentation write `if ip == nil { return }`

Reply:
@manalibhutiyani if you do `if ip == nil { return }` the following part of the code won't have an extra unnecessary indentation.